OpenSim/Simulation/Model/Marker.h

```diff
@@ -57,24 +57,30 @@ class Body;
 //=============================================================================
 // METHODS
 //=============================================================================
-    //--------------------------------------------------------------------------
-    // CONSTRUCTION
-    //--------------------------------------------------------------------------
 public:
     Marker();
     virtual ~Marker();
+    /** Convenience method to get the 'parent_frame' Socket's connectee_name */
+    const std::string& getParentFrameName() const;
 
-    const std::string& getFrameName() const;
+    /** Convenience method to set the 'parent_frame' Socket's connectee_name
+        and to establish the connection during finalizeConnections(). */
+    void setParentFrameName(const std::string& aName);
+    /** Change the parent PhysicalFrame that this marker is attached to. */
+    void changeFrame(const PhysicalFrame& parentFrame);
+    /**  Change the parent PhysicalFrame that this marker is attached to, and 
+         use the state to compute the location in the new parent frame and set
+         its location. */
```

Comment:
I think it would be helpful to explicitly include some variation of the comment that appears in the code: "*Preserve location* means that the Marker is moved to the provided PhysicalFrame without changing the location of the Marker in the inertial reference frame." It is possible to misinterpret the meaning of "preserve location", particularly if the user is unfamiliar with whether the `location` property holds coordinates in the local or Ground frame.